memory/libhugetlbfs.py

```diff
@@ -98,11 +100,37 @@ def setUp(self):
         os.chdir(self.srcdir)
         patch = self.params.get('patch', default='elflink.patch')
         process.run('patch -p1 < %s' % data_dir + '/' + patch, shell=True)
+
+        if (detected_distro.name == "redhat" or
+                detected_distro.name == "fedora"):
+            falloc_patch = 'patch -p1 < %s ' % (
+                os.path.join(data_dir, 'falloc.patch'))
+            process.run(falloc_patch, shell=True)
+
         build.make(self.srcdir, extra_args='BUILDTYPE=NATIVEONLY')
 
+    def fail_check(self, log, check):
```

Comment:
Since this is an enhancement pull request, let me suggest something here. It took some time for me to understand this function. Instead of a `fail_check()` that digs into the log to find the errors, I'd create a log parser so you can easily verify the results inside your `test()`. For this log results, something like this:

```python

def _log_parser(self, log):
    """
    Parses the log, returning a dictionary with the test results.

    Test summary section example:
        ********** TEST SUMMARY
        *                      16M          
        *                      32-bit 64-bit
        *     Total testcases:     0     93  
        *             Skipped:     0      0  
        *                PASS:     0     90  
        *                FAIL:     0      3  
        *    Killed by signal:     0      0  
        *   Bad configuration:     0      0  
        *       Expected FAIL:     0      0  
        *     Unexpected PASS:     0      0  
        * Strange test result:     0      0  
        **********

    Return example:
        {32: {'Bad': 0,
              'Expected': 0,
              'FAIL': 0,
              'Killed': 0,
              'PASS': 0,
              'Skipped': 0,
              'Strange': 0,
              'Total': 0,
              'Unexpected': 0},
         64: {'Bad': 0,
              'Expected': 0,
              'FAIL': 3,
              'Killed': 0,
              'PASS': 90,
              'Skipped': 0,
              'Strange': 0,
              'Total': 93,
              'Unexpected': 0}}
    """
    section = False
    parsed_results = {32: {},64: {}}

    for line in log.splitlines():
        if line == '********** TEST SUMMARY\n':
            section = True
        if line == '**********\n':
            section = False
        if section and  ':' in line:
            key, values = line.split(':')
            parsed_results[32][key.split()[1]] = int(values.split()[0])
            parsed_results[64][key.split()[1]] = int(values.split()[1])

    return parsed_results

def test(self):
    ...
    ...
    results = build.run_make(self.srcdir, extra_args='BUILDTYPE=NATIVEONLY check')
    parsed_results = self._log_parser(results.stdout)
    error = None
    if parsed_results[32]['FAIL']:
        error = "%s tests failed for 32-bit\n" % parsed_results[32]['FAIL'])
    if parsed_results[64]['FAIL']:
        error += "%s tests failed for 64-bit" % parsed_results[64]['FAIL'])
   if error:
        self.fail(error)
```